### PR TITLE
feat(firebase-auth): added a new middleware to verify session with firebase-auth

### PR DESCRIPTION
**Pull request by Code-Hex**
Related: #377

- I have added new middleware that enables authentication and authorization using cookies.
- updated some deps.
  - Hono remains on version 3, but it is compatible with version 4. This is to ensure that users who have not yet updated to v4 can still use this middleware.

### CODE REVIEW

File: packages/firebase-auth/test/index.test.ts [hunk 1]
```diff
@@ -254,6 +257,316 @@ describe('verifyFirebaseAuth middleware', () => {
...
+
+      expect(res).not.toBeNull()
+      expect(res.status).toBe(wantStatus)
+      if (wantStatus === 302) {
+        expect(res.headers.get('location')).toBe(signInPath)
+      }
+    })
+
+    // NOTE(codehex): We can't check this test because https://identitytoolkit.googleapis.com/v1/sessionCookiePublicKeys endpoint
+    // responds with `cache-control: no-cache, no-store, max-age=0, must-revalidate`
```
Comment (Code-Hex) [inline packages/firebase-auth/test/index.test.ts:453]: my TODO 😇

File: packages/firebase-auth/README.md [hunk 2]
```diff
@@ -100,6 +100,168 @@ You can specify a host for the Firebase Auth emulator. This config is mainly use
...
+
+https://hono.dev/middleware/builtin/csrf
+
+**XSS**
+
+An attacker can inject a script and steal JWTs stored in cookies. Set the httpOnly flag on the cookie to prevent access from JavaScript. Additionally, configure "Content Security Policy" (CSP) to prevent unauthorized script execution. It is recommeded to force httpOnly and the functionality here: https://hono.dev/middleware/builtin/secure-headers
+
+**MitM**
+
+If your cookie security settings are inappropriate, there is a risk that your cookies will be stolen by a MitM. Use Samesite (or hono csrf middleware) and `__Secure-` prefix and `__Host-` prefix attributes.
```
Comment (Code-Hex) [inline packages/firebase-auth/README.md:249]: @Datron creates PR to honojs/hono for this.

**Comment by Code-Hex** [comment]
@yusukebe Could you check this PR please 🙏 

I designed the code for this middleware to be as close as possible to what I use in production.

**Review by yusukebe — APPROVED** [review]
LGTM!

**Comment by yusukebe** [comment]
Hi @Code-Hex !

Looks good to me! Can I merge this right now and release it?

**Comment by Code-Hex** [comment]
@yusukebe Sure! Thank you so much!

**Comment by yusukebe** [comment]
@Code-Hex 

Okay! Let's go!

**Pull request merged**